### DISCUSSION TITLE
Where is the output of LLog::fatal(...) and others?

### DISCUSSION THREAD

**Discussion by brianhsu** [CuarzoSoftware/Louvre#26] [Support] [answered]
Hi, I'm experimenting Louvre by go through the [this tutorial](https://cuarzosoftware.github.io/Louvre/md_md_tutorial_03.html).  

The first example code is pretty straightforward. My question regards it is when something go wrong, where will the output of [LLog::fatal](https://cuarzosoftware.github.io/Louvre/class_louvre_1_1_l_log.html#abdee3dc73ec124f69d84a83af3ea90ce) display or stored?

Or more generic, if I have called a method `LLog::log("Hello World")`, where could I find the output?

**Comment by brianhsu** [comment, marked as answer]
OK, I found it. It's basically a printf so it will output to [standard output](https://github.com/CuarzoSoftware/Louvre/blob/a89e9f3fd31373693161c0e21d761530525935d1/src/lib/core/LLog.cpp#L4). 

I will need pipe it into a file by using `tee` or a pipe in my shell, so I could see the logged message.

**Reply by ehopperdietzel** [reply]
Hello, thank you for noticing that. As you mentioned, all messages currently go to stdout. Do you think I should redirect warning, error, and fatal messages to stderr? Or perhaps only error and fatal messages?

**Reply by brianhsu** [reply]
From the naming, I would assume `error` and `fatal` writes to stderr. No so sure about `warning`, I think either `stdout` or `stderr` is fine.

Also, as long as the [document of `LLog`](https://cuarzosoftware.github.io/Louvre/class_louvre_1_1_l_log.html#details) clearly states which output stream these methods write to, I would think this is not a big deal.

**Reply by ehopperdietzel** [reply]
Ok I'll change it, thanks!
